locksmith/src/utils/promises.ts

```diff
@@ -0,0 +1,18 @@
+export const waitOrTimeout = async (
```

Comment:
A simple util to race with a timeout and yield back control to the caller when that happens.